File: tests/api-client/CursosApiClient.ts
import { APIRequestContext, expect } from '@playwright/test';
import { API_BASE_URL } from '@support/env';

export class CursosApiClient {
  constructor(
    private readonly request: APIRequestContext,
    private readonly token?: string
  ) {}

  async buscarPorTecnologia(tecnologia: string) {
    const response = await this.request.get(`${API_BASE_URL}/api/cursos`, {
      headers: this.token ? { Authorization: `Bearer ${this.token}` } : undefined,
      params: { tecnologia },
    });

    expect(response.ok()).toBeTruthy();
    return await response.json();
  }
}

File: playwright-dotnet/PlaywrightDotnet.Tests/ApiClients/CursosApiClient.cs
using System;
using System.Net.Http;
using System.Net.Http.Json;
using System.Text.Json;
using System.Threading.Tasks;

namespace PlaywrightDotnet.Tests.ApiClients
{
    public class CursosApiClient
    {
        private readonly HttpClient client;

        public CursosApiClient(string baseUrl, string? token = null)
        {
            client = new HttpClient { BaseAddress = new Uri(baseUrl) };
            if (!string.IsNullOrEmpty(token))
            {
                client.DefaultRequestHeaders.Authorization = new System.Net.Http.Headers.AuthenticationHeaderValue("Bearer", token);
            }
        }

        public async Task<JsonElement> BuscarPorTecnologiaAsync(string tecnologia)
        {
            var response = await client.GetAsync($"/api/cursos?tecnologia={Uri.EscapeDataString(tecnologia)}");
            response.EnsureSuccessStatusCode();
            using var doc = await JsonDocument.ParseAsync(await response.Content.ReadAsStreamAsync());
            return doc.RootElement.Clone();
        }
    }
}
